cyral/resource_cyral_access_rules.go

```diff
@@ -0,0 +1,292 @@
+package cyral
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cyralinc/terraform-provider-cyral/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type AccessRulesIdentity struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
+}
+
+type AccessRulesConfig struct {
+	AuthorizationPolicyInstanceIDs []string `json:"authorizationPolicyInstanceIDs"`
+}
+
+type AccessRule struct {
+	Identity   *AccessRulesIdentity `json:"identity"`
+	ValidFrom  *string              `json:"validFrom"`
+	ValidUntil *string              `json:"validUntil"`
+	Config     *AccessRulesConfig   `json:"config"`
+}
+
+type AccessRulesResource struct {
+	AccessRules []*AccessRule `json:"accessRules"`
+}
+
+type AccessRuleSet struct {
+	AccessRules []*AccessRule `json:"accessRules"`
+}
+
+type AccessRulesResponse struct {
+	AccessRuleSet AccessRuleSet `json:"accessRuleSet"`
+}
+
+// WriteToSchema is used when reading a resource. It takes whatever the API
+// read call returned, and translates it into the Terraform schema.
+func (arr *AccessRulesResponse) WriteToSchema(d *schema.ResourceData) error {
+	// We'll have to build the access rule set in the format expected by Terraform,
+	// which boils down to doing a bunch of type casts
+	rules := make([]interface{}, 0, len(arr.AccessRuleSet.AccessRules))
+	for _, rule := range arr.AccessRuleSet.AccessRules {
+		m := make(map[string]interface{})
+		m["identity"] = map[string]interface{}{
+			"type": rule.Identity.Type,
+			"name": rule.Identity.Name,
+		}
+		m["validFrom"] = rule.ValidFrom
+		m["validUntil"] = rule.ValidUntil
+		m["config"] = map[string]interface{}{
+			"authorizationPolicyInstanceIDs": rule.Config.AuthorizationPolicyInstanceIDs,
+		}
+
+		rules = append(rules, m)
+	}
+	d.Set("rule", rules)
```

Comment:
```suggestion
	return d.Set("rule", rules)
```

`d.Set` returns an error. Although we don't expect this to err, and we currently ignore this error in a lot of places in our code, I think it is best if we propagate it.

Reply:
Done. Thanks for the suggestion!